src/collection-browser.ts

```diff
@@ -384,44 +386,75 @@ export class CollectionBrowser
   }
 
   private get collectionBrowserTemplate() {
```

Comment:
The old version of this template was getting a bit bloated and hard to read, so I've split it out into several smaller sub-templates that should hopefully be easier to understand.